homeassistant/components/unifi/controller.py

```diff
@@ -375,8 +382,32 @@ async def async_setup(self):
 
         self.config_entry.add_update_listener(self.async_config_entry_updated)
 
+        self._cancel_disconnected_check = async_track_time_interval(
+            self.hass, self._async_check_for_disconnected, CHECK_DISCONNECTED_INTERVAL
+        )
+
         return True
 
+    @callback
+    def add_disconnected_check(self, entity: Entity) -> None:
+        """Add an entity to watch for disconnection."""
+        self._watch_disconnected_entites.append(entity)
+
+    @callback
+    def remove_disconnected_check(self, entity: Entity) -> None:
+        """Remove an entity to watch for disconnection."""
+        self._watch_disconnected_entites.remove(entity)
+
+    @callback
+    def _async_check_for_disconnected(self, *_) -> None:
+        """Check for any devices scheduled to be marked disconnected."""
+        now = dt_util.utcnow()
+
+        for entity in self._watch_disconnected_entites:
+            disconnected_time = entity.disconnected_time
+            if disconnected_time is not None and now > disconnected_time:
+                entity.make_disconnected()
```

Comment:
We don't want to access the Entity directly from outside the entity. We use signals, our dispatch helper, to tell entities what to do outside of the entity platform helper.

Reply:
Addressed in https://github.com/home-assistant/core/pull/45211